seametrics/horizon/README.md

```diff
@@ -0,0 +1,112 @@
+## Horizon Metrics
+
+This metric uses `seametrics.horizon.HorizonMetrics` to calculate the slope and midpoint errors.
+
+## How to Use
+
+To utilize horizon-metrics effectively, start by installing the necessary dependencies using the provided pip command. Once installed, import the evaluate library into your Python environment. Then, use this metric to evaluate your horizon prediction models. Ensure that both ground truth and prediction points are correctly formatted before computing the result. Finally, analyze the computed result to gain insights into the performance of your prediction models.
+
+### Getting Started
+
+To get started with horizon-metrics, make sure you have the necessary dependencies installed. This metric relies on the `seametrics` libraries.
+
+### Installation
+
+```sh
+  pip install evaluate git+https://github.com/SEA-AI/seametrics@develop
+```
+
+### Basic Usage
+
+This is how you can quickly evaluate your horizon prediction models using the horizon-metrics:
+
+##### Use artifical data for testing
+
+```python
+ground_truth_points = [[[0.0, 0.5384765625], [1.0, 0.4931640625]],
+                       [[0.0, 0.53796875], [1.0, 0.4928515625]],
+                       [[0.0, 0.5374609375], [1.0, 0.4925390625]],
+                       [[0.0, 0.536953125], [1.0, 0.4922265625]],
+                       [[0.0, 0.5364453125], [1.0, 0.4919140625]]]
+
+prediction_points = [[[0.0, 0.5428930956049597], [1.0, 0.4642497615378973]],
+                     [[0.0, 0.5428930956049597], [1.0, 0.4642497615378973]],
+                     [[0.0, 0.523573113510805], [1.0, 0.47642688648919496]],
+                     [[0.0, 0.5200016849393765], [1.0, 0.4728554579177664]],
+                     [[0.0, 0.523573113510805], [1.0, 0.47642688648919496]]]
+```
+
+##### Load data from fiftyone
+
+```python
+# Load data from fiftyone
+sequence = "Sentry_2022_11_PROACT_CELADON_7.5M_MOB_2022_11_25_12_29_48"
+dataset_name = "SENTRY_VIDEOS_DATASET_QA"
+sequence_view = fo.load_dataset(dataset_name).match(F("sequence") == sequence)
+sequence_view = sequence_view.select_group_slices("thermal_wide")
+
+# Get the ground truth points
+polylines_gt = sequence_view.values("frames.ground_truth_pl")
+ground_truth_points = [
+    line["polylines"][0]["points"][0] if line is not None else None
+    for line in polylines_gt[0]
+]
+
+# Get the predicted points
+polylines_pred = sequence_view.values(
+    "frames.ahoy-IR-b2-whales__XAVIER-AGX-JP46_pl_TnFoV")
+prediction_points = [
+    line["polylines"][0]["points"][0] if line is not None else None
+    for line in polylines_pred[0]
+```
+
+##### Calculate horizon metrics
+
+```python
+from seametrics.horizon.horizon import HorizonMetrics
+
+# Create the metrics object
+metrics = HorizonMetrics(roll_threshold=0.5, pitch_threshold=0.1, vertical_fov_degrees=25.6, height=512)
```

Comment:
Can we add some info on the README about the available parameters?
Also, I have the following concerns:
- What if we do not pass any of the parameters? will it give only normalised results?
- What if not all the images have the same height and/or vertical FoV?

No need to further change the code, making explicit the info in the README is good enough for now.

Reply:
Added information to the readme on github and on huggingface. 
<img width="867" alt="image" src="https://github.com/SEA-AI/seametrics/assets/29953180/a9921fe8-8a4c-4ddf-bc98-bffd087ac912">


Removed the default parameters for vertical_fov and height to prevent usage without parameters (but maybe different parameters would be needed) leading to wrong metrics without error -> so height and fov are required now:
<img width="835" alt="image" src="https://github.com/SEA-AI/seametrics/assets/29953180/918ac450-c32a-4db1-a72b-542e7f20ab04">